src/brsTypes/components/RoSGNode.ts

```diff
@@ -0,0 +1,222 @@
+import { BrsValue, ValueKind, BrsString, BrsInvalid, BrsBoolean } from "../BrsType";
+import { BrsComponent, BrsIterable } from "./BrsComponent";
+import { BrsType } from "..";
+import { Callable, StdlibArgument } from "../Callable";
+import { Interpreter } from "../../interpreter";
+import { Int32 } from "../Int32";
+import { RoAssociativeArray } from "./RoAssociativeArray";
+import { RoArray } from "./RoArray";
+import { AAMember } from "./RoAssociativeArray";
+
+class Field {
+    constructor(readonly type: string, readonly alwaysNotify: boolean) {}
+}
+
+export class RoSGNode extends BrsComponent implements BrsValue, BrsIterable {
+    readonly kind = ValueKind.Object;
+    private elements = new Map<string, BrsType>();
+    private fields = new Map<string, Field>();
+
+    constructor(elements: AAMember[], readonly type: string = "Node") {
+        super("roSGNode");
+        elements.forEach(member =>
+            this.elements.set(member.name.value.toLowerCase(), member.value)
+        );
+
+        this.registerMethods([
+            // ifAssociativeArray methods
+            this.clear,
+            this.delete,
+            this.addreplace,
+            this.count,
+            this.doesexist,
+            this.append,
+            this.keys,
+            this.items,
+            this.lookup,
+        ]);
+    }
+
+    toString(parent?: BrsType): string {
+        let componentName = "roSGNode:" + this.type;
+
+        if (parent) {
+            return `<Component: ${componentName}>`;
+        }
+
+        return [
+            `<Component: ${componentName}> =`,
+            "{",
+            ...Array.from(this.elements.keys()).map(
+                key => `    ${key}: ${this.elements.get(key)!.toString(this as any)}`
```

Comment:
The "as any" is to avoid a type error as this class isn't a BrsType, I guess.  Not sure how to improve this.

Reply:
Yep, I think that might have been solved by a newer typescript version?